drivers/include/periph/freqm.h

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 2023 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_periph_freqm FREQM
+ * @ingroup     drivers_periph
+ * @brief       FREQM peripheral driver interface
+ *
+ * This interface allows to configure and use the Frequency Meter (FREQM)
+ * peripheral.
+ *
+ * The Frequency Meter uses the frequency of a known reference clock to
+ * determine the frequency of a signal connected via GPIO.
+ *
+ * @{
+ *
+ * @file
+ * @brief       FREQM peripheral driver interface definitions
+ *
+ * @author      Urs Gompper <urs.gompper@ml-pa.com>
+ */
+
+#ifndef PERIPH_FREQM_H
+#define PERIPH_FREQM_H
+
+#include <mutex.h>
+
+#include "periph_cpu.h"
+#include "periph/gpio.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef void (*freqm_callback_t)(void *context);
```

Comment:
```suggestion
typedef void (*freqm_callback_t)(uint32_t result, void *context);
```

any reason why not directly pass the frequency?

Reply:
Maybe I'm overseeing something, but if I want to pass it, then I have to read/calculate the frequency inside the ISR (which I tried to avoid)?

Reply:
AFAIU you get this interrupt when the measurement is ready, so reading doesn't have to wait for anything.
And the only 'expensive' thing I see in `freqm_frequency_get()` is the division, but that really isn't something we need to worry about.

Reply:
Okay, then I would also pass the `overflow_condition` of the value counter, to indicate if a shorter measurement time or a faster reference clock has to be configured.